kernel/eraseconv.ml

```diff
@@ -0,0 +1,75 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Constr
+open Conversion
+
+(** The erase evaluation function can either return an evaluated term or an error if the
+    evaluated term cannot be read back *)
+type erase_evaluation_function =
+  Environ.env -> Constr.t -> types -> (Constr.t, unit) result
```

Comment:
how about allow returning a message (Pp.t) in the error case?

Reply:
Indeed that will be done :)